app/controllers/users_controller.rb

```diff
@@ -9,6 +9,7 @@ def destroy
 
   def destroy_user
     user = current_user || User.find_by(confirmation_token: session[:user_confirmation_token])
+    session.delete(:user_confirmation_token)
```

Comment:
It was an honest question initially -- if we are deleting a user, should we also delete the related session. Every visitor has a session, whether they are authenticated or not. If just deleting the `user_confirmation_token` does the trick in this case, then that seems fine.